db/migrate/20230906165200_migrate_imported_variants_data.rb

```diff
@@ -0,0 +1,181 @@
+
+    def variant_display_as?
+      respond_to?(:variant) && variant.present? &&
+        variant.respond_to?(:display_as) && variant.display_as.present?
+    end
+
+    def product_and_full_name
+      return product.name if full_name.blank?
+      return "#{product.name} - #{full_name}" unless full_name.start_with?(product.name)
+
+      full_name
+    end
+
+    # Used like "product.name - full_name", preferably using product_and_full_name method above.
+    # This returns, for a product with name "Bread":
+    #     Bread - 1kg                     # if display_name blank
+    #     Bread - Spelt Sourdough, 1kg    # if display_name is "Spelt Sourdough, 1kg"
+    #     Bread - 1kg Spelt Sourdough     # if unit_to_display is "1kg Spelt Sourdough"
+    # if display_name is "Spelt Sourdough" and unit_to_display is "1kg"
+    #     Bread - Spelt Sourdough (1kg)
+    def full_name
+      return unit_to_display if display_name.blank?
+      return display_name    if display_name.downcase.include? unit_to_display.downcase
+      return unit_to_display if unit_to_display.downcase.include? display_name.downcase
+
+      "#{display_name} (#{unit_to_display})"
+    end
+
+    def name_to_display
+      return product.name if display_name.blank?
+
+      display_name
+    end
+
+    def unit_to_display
+      return display_as if has_attribute?(:display_as) && display_as.present?
+      return variant.display_as if variant_display_as?
+
+      options_text.to_s
+    end
+
+    def assign_units
+      assign_attributes(unit_value_attributes)
+    end
+
+    def update_units
+      update_columns(unit_value_attributes)
+    end
+
+    def unit_value_attributes
+      units = { unit_presentation: option_value_name }
+      units.merge!(variant_unit: product.variant_unit) if has_attribute?(:variant_unit)
+      units
+    end
+
+    def weight_from_unit_value
+      (unit_value || 0) / 1000 if product.variant_unit == 'weight'
+    end
+
+    private
+
+    def option_value_name
+      return display_as if has_attribute?(:display_as) && display_as.present?
+
+      OptionValueNamer.new(self).name
+    end
+  end
+
+  class Variant < ActiveRecord::Base
+    include VariantAndLineItemNaming
+
+    belongs_to :product
+
+    self.table_name = "spree_variants"
+  end
+
+  class Product < ActiveRecord::Base
+    has_many :variants
+
+    self.table_name = "spree_products"
+  end
+
+  def up
+    migrate_variant_unit
+
+    Variant.where(unit_presentation: "").where.not(import_date: nil).each do |variant|
+      variant.update_columns(
+        variant.unit_value_attributes.merge(updated_at: Time.zone.now)
+      )
+    end
+  end
+
+  private
+
+  def migrate_variant_unit
+    ActiveRecord::Base.connection.execute(<<-SQL
+      UPDATE spree_variants
+      SET variant_unit = spree_products.variant_unit
+      FROM spree_products
+      WHERE spree_variants.product_id = spree_products.id
+        AND spree_variants.variant_unit IS NULL
+    SQL
+    )
+  end
```

Comment:
This doesn't look directly related to the importer issue. Is this necessary because we didn't do it as part of the earlier refactor? 

In any case, this potentially affects a lot of records so would you be able to add some detail about this in the PR description please?

Reply:
This was done before in 35b41fc7f but needs repeating now because the code was faulty and didn't always set that value.